apps/frontend/src/app/Database/DataManagers/CharacterData.ts

```diff
@@ -34,17 +34,17 @@ export class CharacterDataManager extends DataManager<CharacterKey, "characters"
 
     if (!allHitModes.includes(hitMode)) hitMode = "avgHit"
     if (reaction && !allAmpReactions.includes(reaction as typeof allAmpReactions[number]) && !allAdditiveReactions.includes(reaction as typeof allAdditiveReactions[number])) reaction = undefined
-    if (infusionAura!=="" && !allInfusionAuraElements.includes(infusionAura as InfusionAuraElements)) infusionAura = ""
+    if (infusionAura !== "" && !allInfusionAuraElements.includes(infusionAura as InfusionAuraElements)) infusionAura = ""
     if (typeof constellation !== "number" && constellation < 0 && constellation > 6) constellation = 0
 
     const { level, ascension } = validateLevelAsc(rawLevel, rawAscension)
 
     if (typeof talent !== "object") talent = { auto: 1, skill: 1, burst: 1 }
     else {
       let { auto, skill, burst } = talent
-      if (typeof auto !== "number" || auto < 1 || auto > 15) auto = 1
-      if (typeof skill !== "number" || skill < 1 || skill > 15) skill = 1
-      if (typeof burst !== "number" || burst < 1 || burst > 15) burst = 1
+      auto = typeof auto !== "number" ? 1 : clamp(auto, 1, 10)
+      skill = typeof skill !== "number" ? 1 : clamp(skill, 1, 10)
+      burst = typeof burst !== "number" ? 1 : clamp(burst, 1, 10)

```

Comment:
Should we add logic based on ascension as well? No tracking issue at the moment, but its a known bug

Reply:
i thought about this, the issue is that if level/asc/talent is out of whack, which is the true value that everything is validated against? i'd argue the truth is level, which validates asc, which validates talent.

Reply:
Yea its not often for those things to be out of sync so asc or level is fine to validate first, and validate talent as a final step. Pretty rare that would affect anyone so no big deal in having it be slightly arbitrary imo